server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -266,13 +256,12 @@ public void rollGeneration() throws IOException {
         if (current.totalOperations() == 0 && primaryTermSupplier.getAsLong() == current.getPrimaryTerm()) {
             return;
         }
-        if (prepareForUpload(null)) {
-            performUpload(primaryTermSupplier.getAsLong(), null);
-        }
+        prepareAndUpload(primaryTermSupplier.getAsLong(), null);
     }
 
-    private boolean prepareForUpload(Long generation) throws IOException {
-        try (Releasable ignored = writeLock.acquire()) {
+    private boolean prepareAndUpload(Long primaryTerm, Long generation) throws IOException {
+        Releasable uploadReadLock = null;
+        try (ReleasableLock updateWriteLock = writeLock.acquire()) {
```

Comment:
here as well , we don't need to create `updateWriteLock` . `acquire()` returns same object . So we can use `writeLock` itself.

Reply:
the try argument needs to be a final object, hence we need to assign it to a new variable.